src/MultiSelect.js

```diff
@@ -1,51 +1,63 @@
+// @flow
 import React from 'react';
 
-class MultiSelectOption extends React.Component {
+type MultiSelectOptionProps = {
+    label: string,
+    value: string
+};
+
+class MultiSelectOption extends React.Component<MultiSelectOptionProps, {}> {
     render() {
         return(
             <option value={this.props.value}>{this.props.label}</option>
         );
     }
 }
 
-export default class MultiSelect extends React.Component {
-    constructor(props) {
+type OptionDef = {
+    value: string,
+    label?: string
+}
+
+type MultiSelectProps = {
+    title: string,
+    selectedValues?: Array<Object>,
+    stateChangeCallbacks?: Array<Function>,
+    defaultValue?: any,
+    options: { [string]: OptionDef }
+}
+
+type MultiSelectState = {
+    selectedValues: Array<Object>
+};
+
+export default class MultiSelect extends React.Component<MultiSelectProps, MultiSelectState> {
+    constructor(props: MultiSelectProps) {
         super(props);
-<<<<<<< HEAD
         this.state = {
             selectedValues: props.selectedValues || []
         }
-=======
-        this.state={
-            selectedValues: props.selectedValues || []
-        }
-
-        this.handleChange = this.handleChange.bind(this);
->>>>>>> master
     }
 
     // Apparently react can't track the selected attribute of individual options, there's an explicit warning against it.
     // It seems like you're supposed to listen to changes on select elements instead.
-<<<<<<< HEAD
-    handleChange = (event) => {
-=======
-    handleChange(event) {
->>>>>>> master
-        let selectedValueMap = {};
-        for (let i = 0; i < event.target.options.length; i++) {
-            let singleOption = event.target.options[i];
+    handleChange = (event: SyntheticEvent<HTMLSelectElement>) => {
+        const selectedValueMap = { };
+        for (let i = 0; i < event.currentTarget.options.length; i++) {
+            const singleOption: HTMLOptionElement = event.currentTarget.options[i];
             if (singleOption.selected) {
                 selectedValueMap[singleOption.value] = true;
             }
         }
 
-        let selectedValues = Object.keys(selectedValueMap);
+        const selectedValues: Array<string> = Object.keys(selectedValueMap);
         this.setState({
             selectedValues: selectedValues
         });
 
+        // TODO: Discuss why type coverage is not picked up based on the annotations above.
         if (this.props.stateChangeCallbacks) {
-            this.props.stateChangeCallbacks.forEach((callback) => {
+            this.props.stateChangeCallbacks.forEach((callback: Function) => {
```

Comment:
In the coding environment, we specify function type as an anonymous arrow function like 'callback: () => void'; you can also specify input parameter type to the function like, 'callback: (index: number) => void'.

Reply:
Thanks, that fixes the last coverage warning.